File: 프로그래머스/2/42842. 카펫/카펫.py
from itertools import combinations

def check_area(x, y, yellow):
    return True if (x * y) + (-2) * (x + y) + 4 == yellow else False

def solution(brown, yellow):
    area = brown + yellow
    x = area // 3
    y = 3
    
    while x >= y:
        if area % x == 0:
            y = area // x
            
            if check_area(x, y, yellow):
                return [x, y]
        
        x -= 1
        

File: 프로그래머스/2/42842. 카펫/카펫.kt
import kotlin.math.sqrt

class Solution {
    fun solution(brown: Int, yellow: Int): IntArray {
        val area = brown + yellow
        val maxH = sqrt(area.toDouble()).toInt()
        
        for (h in 3..maxH) {
            if (area % h != 0) continue
            
            val w = area / h
            
            if (2 * (w + h) - 4 == brown && (w - 2) * (h - 2) == yellow) {
                return intArrayOf(w, h)
            }
        }
        
        error("종료")
    }
}